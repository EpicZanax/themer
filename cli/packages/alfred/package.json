{
  "name": "@themerdev/alfred",
  "version": "2.0.0",
  "description": "An Alfred.app theme generator for themer.",
  "main": "lib/index.mjs",
  "engines": {
    "node": ">=18"
  },
  "scripts": {
    "prepublishOnly": "cp ../../../LICENSE.md ./",
    "test": "vitest"
  },
  "author": "mjswensen",
  "license": "MIT",
  "files": [
    "/lib/index.mjs"
  ],
  "devDependencies": {
    "@themerdev/colors-default": "^2.0.0",
    "vitest": "^0.25.1"
  },
  "peerDependencies": {
    "themer": "^4"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/themerdev/themer.git",
    "directory": "cli/packages/alfred"
  },
  "keywords": [
    "themer",
    "alfred",
    "theme"
  ],
  "bugs": {
    "url": "https://github.com/themerdev/themer/issues"
  },
  "homepage": "https://github.com/themerdev/themer/tree/main/cli/packages/alfred#readme"
}
