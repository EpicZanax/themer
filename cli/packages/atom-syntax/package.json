{
  "name": "@themerdev/atom-syntax",
  "version": "3.0.0",
  "description": "Atom syntax template for themer.",
  "main": "lib/index.mjs",
  "engines": {
    "node": ">=18"
  },
  "author": "mjswensen",
  "license": "MIT",
  "files": [
    "/lib/index.mjs"
  ],
  "scripts": {
    "prepublishOnly": "cp ../../../LICENSE.md ./",
    "test": "vitest"
  },
  "devDependencies": {
    "@themerdev/colors-default": "^2.0.0",
    "less": "^4.1.1",
    "vitest": "^0.25.1"
  },
  "peerDependencies": {
    "themer": "^4"
  },
  "repository": {
    "type": "git",
    "url": "git+ssh://git@github.com/themerdev/themer.git",
    "directory": "cli/packages/atom-syntax"
  },
  "bugs": {
    "url": "https://github.com/themerdev/themer/issues"
  },
  "homepage": "https://github.com/themerdev/themer/tree/main/cli/packages/atom-syntax#readme",
  "keywords": [
    "themer",
    "atom",
    "syntax",
    "theme"
  ]
}
