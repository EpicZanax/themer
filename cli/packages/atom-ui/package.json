{
  "name": "@themerdev/atom-ui",
  "version": "2.0.0",
  "description": "An Atom UI theme generator for themer.",
  "main": "lib/index.mjs",
  "engines": {
    "node": ">=18"
  },
  "scripts": {
    "prepublishOnly": "cp ../../../LICENSE.md ./",
    "test": "vitest"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/themerdev/themer.git",
    "directory": "cli/packages/atom-ui"
  },
  "keywords": [
    "themer",
    "atom",
    "ui",
    "theme"
  ],
  "author": "mjswensen",
  "license": "MIT",
  "files": [
    "/lib/index.mjs",
    "/lib/templates/baseLess.mjs",
    "/lib/templates/buttonsLess.mjs",
    "/lib/templates/editorLess.mjs",
    "/lib/templates/findAndReplaceLess.mjs",
    "/lib/templates/gitLess.mjs",
    "/lib/templates/indexLess.mjs",
    "/lib/templates/keyBindingLess.mjs",
    "/lib/templates/listsLess.mjs",
    "/lib/templates/packageJson.mjs",
    "/lib/templates/panelsLess.mjs",
    "/lib/templates/progressLess.mjs",
    "/lib/templates/sitesLess.mjs",
    "/lib/templates/statusBarLess.mjs",
    "/lib/templates/tabsLess.mjs",
    "/lib/templates/textLess.mjs",
    "/lib/templates/tooltipsLess.mjs",
    "/lib/templates/treeViewLess.mjs",
    "/lib/templates/uiVariablesLess.mjs"
  ],
  "bugs": {
    "url": "https://github.com/themerdev/themer/issues"
  },
  "homepage": "https://github.com/themerdev/themer/tree/main/cli/packages/atom-ui#readme",
  "devDependencies": {
    "@themerdev/colors-default": "^2.0.0",
    "less": "^4.1.1",
    "lodash": "^4.17.4",
    "vitest": "^0.25.1"
  },
  "peerDependencies": {
    "themer": "^4"
  }
}
