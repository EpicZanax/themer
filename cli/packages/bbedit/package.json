{
  "name": "@themerdev/bbedit",
  "version": "2.0.0",
  "description": "A BBEdit theme generator for themer.",
  "main": "lib/index.mjs",
  "engines": {
    "node": ">=18"
  },
  "scripts": {
    "prepublishOnly": "cp ../../../LICENSE.md ./",
    "test": "vitest"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/themerdev/themer.git",
    "directory": "cli/packages/bbedit"
  },
  "keywords": [
    "themer",
    "bbedit",
    "theme"
  ],
  "author": "mjswensen",
  "license": "MIT",
  "files": [
    "/lib/index.mjs"
  ],
  "bugs": {
    "url": "https://github.com/themerdev/themer/issues"
  },
  "homepage": "https://github.com/themerdev/themer/tree/main/cli/packages/bbedit#readme",
  "dependencies": {
    "color": "^3.0.0",
    "lodash": "^4.17.4",
    "xml": "^1.0.1"
  },
  "devDependencies": {
    "@themerdev/colors-default": "^2.0.0",
    "vitest": "^0.25.1"
  },
  "peerDependencies": {
    "themer": "^4"
  }
}
