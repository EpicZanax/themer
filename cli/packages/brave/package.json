{
  "name": "@themerdev/brave",
  "version": "2.0.0",
  "description": "A Brave theme generator for themer.",
  "main": "lib/index.mjs",
  "engines": {
    "node": ">=18"
  },
  "scripts": {
    "prepublishOnly": "cp ../../../LICENSE.md ./",
    "test": "vitest"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/themerdev/themer.git",
    "directory": "cli/packages/brave"
  },
  "keywords": [
    "themer",
    "brave",
    "theme"
  ],
  "author": "mjswensen",
  "license": "MIT",
  "files": [
    "/lib/index.mjs"
  ],
  "bugs": {
    "url": "https://github.com/themerdev/themer/issues"
  },
  "homepage": "https://github.com/themerdev/themer/tree/main/cli/packages/brave#readme",
  "dependencies": {
    "@themerdev/chrome": "^2"
  },
  "devDependencies": {
    "@themerdev/colors-default": "^2.0.0",
    "vitest": "^0.25.1"
  },
  "peerDependencies": {
    "themer": "^4"
  }
}
