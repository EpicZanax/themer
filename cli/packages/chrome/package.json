{
  "name": "@themerdev/chrome",
  "version": "2.0.0",
  "description": "A Chrome theme generator for themer.",
  "main": "lib/index.mjs",
  "engines": {
    "node": ">=18"
  },
  "scripts": {
    "prepublishOnly": "cp ../../../LICENSE.md ./",
    "test": "vitest"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/themerdev/themer.git",
    "directory": "cli/packages/chrome"
  },
  "keywords": [
    "themer",
    "chrome",
    "theme"
  ],
  "author": "mjswensen",
  "license": "MIT",
  "files": [
    "/lib/index.mjs"
  ],
  "bugs": {
    "url": "https://github.com/themerdev/themer/issues"
  },
  "homepage": "https://github.com/themerdev/themer/tree/main/cli/packages/chrome#readme",
  "dependencies": {
    "color": "^3.0.0",
    "lodash": "^4.17.4"
  },
  "devDependencies": {
    "@themerdev/colors-default": "^2.0.0",
    "vitest": "^0.25.1"
  },
  "peerDependencies": {
    "themer": "^4"
  }
}
