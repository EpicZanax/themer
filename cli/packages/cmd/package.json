{
  "name": "@themerdev/cmd",
  "version": "2.0.0",
  "description": "A CMD.exe theme generator for themer.",
  "main": "lib/index.mjs",
  "engines": {
    "node": ">=18"
  },
  "scripts": {
    "prepublishOnly": "cp ../../../LICENSE.md ./",
    "test": "vitest"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/themerdev/themer.git",
    "directory": "cli/packages/cmd"
  },
  "keywords": [
    "themer",
    "cmd",
    "cmd.exe",
    "theme"
  ],
  "author": "mjswensen",
  "license": "MIT",
  "files": [
    "/lib/index.mjs"
  ],
  "bugs": {
    "url": "https://github.com/themerdev/themer/issues"
  },
  "homepage": "https://github.com/themerdev/themer/tree/main/cli/packages/cmd#readme",
  "dependencies": {
    "lodash-es": "^4.17.21"
  },
  "devDependencies": {
    "@themerdev/colors-default": "^1.0.11",
    "vitest": "^0.25.1"
  },
  "peerDependencies": {
    "themer": "^4"
  }
}
