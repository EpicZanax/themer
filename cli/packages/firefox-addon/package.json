{
  "name": "@themerdev/firefox-addon",
  "version": "2.0.0",
  "description": "A Firefox theme generator for themer.",
  "main": "lib/index.mjs",
  "engines": {
    "node": ">=18"
  },
  "scripts": {
    "prepublishOnly": "cp ../../../LICENSE.md ./",
    "test": "vitest"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/themerdev/themer.git",
    "directory": "cli/packages/firefox-addon"
  },
  "keywords": [
    "themer",
    "firefox",
    "theme"
  ],
  "author": "mjswensen",
  "license": "MIT",
  "files": [
    "/lib/index.mjs"
  ],
  "bugs": {
    "url": "https://github.com/themerdev/themer/issues"
  },
  "homepage": "https://github.com/themerdev/themer/tree/main/cli/packages/firefox-addon#readme",
  "dependencies": {
    "color": "^4.2.3",
    "lodash-es": "^4.17.21"
  },
  "devDependencies": {
    "@themerdev/colors-default": "^1.0.11",
    "vitest": "^0.25.1"
  },
  "peerDependencies": {
    "themer": "^4"
  }
}
