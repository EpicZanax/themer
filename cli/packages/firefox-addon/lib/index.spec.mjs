import { pickBy } from 'lodash-es';
import { render, renderInstructions } from './index.mjs';
import { colors } from '@themerdev/colors-default';
import { describe, expect, it } from 'vitest';
import { version } from '../package.json' assert { type: 'json' };
import { dirname } from 'node:path';

describe('Firefox theme generator', () => {
  it('should render a directory containing a manifest.json', async () => {
    const files = await Promise.all(render(colors));
    const dirnames = files.map((file) => dirname(file.name));
    expect(dirnames.length).toBe(2);
    expect(dirnames).toContain('Themer Light');
    expect(dirnames).toContain('Themer Dark');
  });
  it('should render properly formatted manifest files', async () => {
    const files = await Promise.all(render(colors));
    files.forEach((file) => {
      const contents = JSON.parse(file.contents);
      expect(contents.version).toBe(version);
      expect(pickBy(contents, (_, key) => key !== 'version')).toMatchSnapshot();
    });
  });
  it('should provide installation instructions', async () => {
    const files = await Promise.all(render(colors));
    const instructions = renderInstructions(files.map(({ name }) => name));
    expect(instructions).toMatchSnapshot();
  });
});
