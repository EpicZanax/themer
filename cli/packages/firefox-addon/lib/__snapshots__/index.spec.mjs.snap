// Vitest Snapshot v1

exports[`Firefox theme generator > should provide installation instructions 1`] = `
"
To use the generated extension package, the code will need to be packaged up and signed by Mozilla.

To package the code in preparation for submission, the \`web-ext\` tool can be used:

    npx web-ext build --source-dir 'Themer Dark' # or 'Themer Light'

Then the package can be submitted to Mozilla for review in the [Add-on Developer Hub](https://addons.mozilla.org/en-US/developers/addon/submit/distribution).

Learn more about Firefox themes from [extensionworkshop.com](https://extensionworkshop.com/documentation/themes/)

To theme Firefox without the need to create a developer account and go through the extension review process, see themer's integration with [Firefox Color](https://color.firefox.com).
  "
`;

exports[`Firefox theme generator > should render properly formatted manifest files 1`] = `
{
  "manifest_version": 2,
  "name": "Themer Dark",
  "theme": {
    "colors": {
      "bookmark_text": [
        255,
        252,
        255,
      ],
      "frame": [
        40,
        38,
        41,
      ],
      "frame_inactive": [
        71,
        66,
        71,
      ],
      "icons": [
        255,
        252,
        255,
      ],
      "icons_attention": [
        255,
        252,
        255,
      ],
      "ntp_background": [
        71,
        66,
        71,
      ],
      "ntp_text": [
        255,
        252,
        255,
      ],
      "popup": [
        71,
        66,
        71,
      ],
      "popup_border": [
        71,
        66,
        71,
      ],
      "popup_highlight": [
        224,
        220,
        224,
      ],
      "popup_highlight_text": [
        101,
        96,
        102,
      ],
      "popup_text": [
        255,
        252,
        255,
      ],
      "sidebar": [
        71,
        66,
        71,
      ],
      "sidebar_border": [
        71,
        66,
        71,
      ],
      "sidebar_highlight": [
        101,
        96,
        102,
      ],
      "sidebar_highlight_text": [
        224,
        220,
        224,
      ],
      "sidebar_text": [
        224,
        220,
        224,
      ],
      "tab_background_text": [
        255,
        252,
        255,
      ],
      "tab_loading": [
        255,
        252,
        255,
      ],
      "tab_text": [
        255,
        252,
        255,
      ],
      "toolbar": [
        71,
        66,
        71,
      ],
      "toolbar_bottom_separator": [
        71,
        66,
        71,
      ],
      "toolbar_field": [
        71,
        66,
        71,
      ],
      "toolbar_field_border": [
        71,
        66,
        71,
      ],
      "toolbar_field_border_focus": [
        101,
        96,
        102,
      ],
      "toolbar_field_highlight_text": [
        101,
        96,
        102,
      ],
      "toolbar_field_separator": [
        71,
        66,
        71,
      ],
      "toolbar_field_text": [
        255,
        252,
        255,
      ],
      "toolbar_field_text_focus": [
        255,
        252,
        255,
      ],
    },
  },
}
`;

exports[`Firefox theme generator > should render properly formatted manifest files 2`] = `
{
  "manifest_version": 2,
  "name": "Themer Light",
  "theme": {
    "colors": {
      "bookmark_text": [
        40,
        38,
        41,
      ],
      "frame": [
        224,
        220,
        224,
      ],
      "frame_inactive": [
        255,
        252,
        255,
      ],
      "icons": [
        40,
        38,
        41,
      ],
      "icons_attention": [
        40,
        38,
        41,
      ],
      "ntp_background": [
        255,
        252,
        255,
      ],
      "ntp_text": [
        40,
        38,
        41,
      ],
      "popup": [
        255,
        252,
        255,
      ],
      "popup_border": [
        255,
        252,
        255,
      ],
      "popup_highlight": [
        71,
        66,
        71,
      ],
      "popup_highlight_text": [
        193,
        188,
        194,
      ],
      "popup_text": [
        40,
        38,
        41,
      ],
      "sidebar": [
        255,
        252,
        255,
      ],
      "sidebar_border": [
        255,
        252,
        255,
      ],
      "sidebar_highlight": [
        193,
        188,
        194,
      ],
      "sidebar_highlight_text": [
        71,
        66,
        71,
      ],
      "sidebar_text": [
        71,
        66,
        71,
      ],
      "tab_background_text": [
        40,
        38,
        41,
      ],
      "tab_loading": [
        40,
        38,
        41,
      ],
      "tab_text": [
        40,
        38,
        41,
      ],
      "toolbar": [
        255,
        252,
        255,
      ],
      "toolbar_bottom_separator": [
        255,
        252,
        255,
      ],
      "toolbar_field": [
        255,
        252,
        255,
      ],
      "toolbar_field_border": [
        255,
        252,
        255,
      ],
      "toolbar_field_border_focus": [
        193,
        188,
        194,
      ],
      "toolbar_field_highlight_text": [
        193,
        188,
        194,
      ],
      "toolbar_field_separator": [
        255,
        252,
        255,
      ],
      "toolbar_field_text": [
        40,
        38,
        41,
      ],
      "toolbar_field_text_focus": [
        40,
        38,
        41,
      ],
    },
  },
}
`;
